pennylane/_device.py

```diff
@@ -759,7 +758,12 @@ def batch_transform(self, circuit: QuantumTape):
             isinstance(obs, (Hamiltonian, LinearCombination)) for obs in circuit.observables
         )
         expval_sum_in_obs = any(
-            isinstance(m.obs, Sum) and isinstance(m, ExpectationMP) for m in circuit.measurements
+            (
+                isinstance(m.obs, Sum)
+                or (isinstance(m.obs, Prod) and isinstance(m.obs.simplify(), Sum))
```

Comment:
At this point why not include SProd as well?

Reply:
Couldnt the same be said about a sum?

That being said, not sure this is necessary here :)